pkg/reader/ccip.go

```diff
@@ -708,17 +695,36 @@ func (r *ccipChainReader) DiscoverContracts(
 	myChains := maps.Keys(r.contractReaders)
 
 	// Read onRamps for FeeQuoter in DynamicConfig.
-	dynamicConfigs, err := r.getOnRampDynamicConfigs(ctx, myChains)
-	if err != nil {
-		r.lggr.Infow("unable to lookup source fee quoters, this is expected during initialization", "err", err)
+	{
+		dynamicConfigs, err := r.getOnRampDynamicConfigs(ctx, myChains)
+		if err != nil {
+			r.lggr.Infow("unable to lookup source fee quoters, this is expected during initialization", "err", err)
 
-		// ErrNoBindings is an allowable error.
-		if !errors.Is(err, contractreader.ErrNoBindings) {
-			return nil, fmt.Errorf("unable to lookup source fee quoters (onRamp dynamic config): %w", err)
+			// ErrNoBindings is an allowable error.
+			if !errors.Is(err, contractreader.ErrNoBindings) {
+				return nil, fmt.Errorf("unable to lookup source fee quoters (onRamp dynamic config): %w", err)
+			}
```

Comment:
I'm confused by this comment, its an allowable error but we still return an error wrapping `ErrNoBindings`, which is not checked in the discovery processor. Is this expected to only be temporary?

Reply:
Updated control flow to avoid the nested condition:
```
		if errors.Is(err, contractreader.ErrNoBindings) {
			// ErrNoBindings is an allowable error.
			r.lggr.Infow("unable to lookup source routers, this is expected during initialization", "err", err)
		} else if err != nil {
			return nil, fmt.Errorf("unable to lookup source routers (onRamp dest chain config): %w", err)
		} else {
			for chain, cfg := range destChainConfig {
				resp = resp.Append(consts.ContractNameRouter, chain, cfg.Router)
			}
		}
```

Reply:
Ah shoot I totally didn't read the `!` haha

Reply:
I like the new control flow! LGTM